src/main/java/org/sunbird/profile/service/ProfileServiceImpl.java

```diff
@@ -230,6 +232,16 @@ public List<String> approvalFields(String AuthToken,String XAuthToken){
         }
     }
 
+    public void ValidateDepartment(String existingDept, Map<String, Object> RequestProfile) throws Exception {
+        if (RequestProfile.containsKey(Constants.EMPLOYMENTDETAILS)) {
+            Map<String, Object> empDetails = (Map<String, Object>) RequestProfile.get(Constants.EMPLOYMENTDETAILS);
+            String requestDeptName = (String) empDetails.get(Constants.DEPARTMENTNAME);
+            if (!requestDeptName.equals(existingDept)) {
```

Comment:
possible NPE in this case...
if condition should be ->
if(!existingDept.equalsIgnoreCase(requestDeptName))